components/map/MapEvents.tsx

```diff
@@ -21,32 +21,49 @@ export default function MapEvents({ eventList }: Props) {
   const { current: mapRef } = useMap();
   const [popup, setPopup] = useState<Popup | null>(null);
 
-  const geoJsonData = useMemo<FeatureCollection>(
+  const getGeoJsonData = useCallback<() => FeatureCollection>(
     () => ({
       type: 'FeatureCollection',
-      features: eventList.map((event) => ({
-        type: 'Feature',
-        properties: {
-          cluster: false,
-          ...event,
-          start_time: new Date(event.occurrences[0].start).toLocaleTimeString('en-GB', {
-            hour: '2-digit',
-            minute: '2-digit',
-          }),
-          end_time: new Date(event.occurrences[0].end).toLocaleTimeString('en-GB', {
-            hour: '2-digit',
-            minute: '2-digit',
-          }),
-        },
-        geometry: {
-          type: 'Point',
-          coordinates: [event.longitude, event.latidute],
-        },
-      })),
+      features: eventList.map((event) => {
+        const foundOccurrence = event.occurrences.find(
+          (occurrence) => occurrence.end.getTime() > Date.now()
+        );
+
+        return {
+          type: 'Feature',
+          properties: {
+            cluster: false,
+            ...event,
+            start_time: foundOccurrence?.start.toLocaleTimeString('en-GB', {
+              hour: '2-digit',
+              minute: '2-digit',
+            }),
+            end_time: foundOccurrence?.end.toLocaleTimeString('en-GB', {
+              hour: '2-digit',
+              minute: '2-digit',
+            }),
+          },
+          geometry: {
+            type: 'Point',
+            coordinates: [event.longitude, event.latidute],
+          },
+        };
+      }),
     }),
     [eventList]
   );
 
+  const [geoJsonData, setGeoJsonData] = useState<FeatureCollection>(getGeoJsonData());
+
+  useEffect(() => {
+    setGeoJsonData(getGeoJsonData());
+    const interval = setInterval(() => {
+      setGeoJsonData(getGeoJsonData());
+    }, 60000);
```

Comment:
Zamiast setInterval, powinno być memo z `floor(Date.now() / (1000 * 60))`

Reply:
Moim zdaniem i tak to nie będzie działać. Co jak komponent nie będzie renderowany przez dłuższy czas?